src/test/java/com/redhat/rcm/offliner/ftest/SingleFoloRecordDownloadFTest.java

```diff
@@ -90,11 +88,21 @@ public void run()
         // run `new Main(opts).run()` and return the Main instance so we can query it for errors, etc.
         Main finishedMain = run( opts );
 
-        assertThat( "Wrong number of downloads logged. Should have been 1.", finishedMain.getDownloaded(), equalTo( 1 ) );
+        assertThat( "Wrong number of downloads logged. Should have been 1.", finishedMain.getDownloaded(),
+                    equalTo( 1 ) );
         assertThat( "Errors should be empty!", finishedMain.getErrors().isEmpty(), equalTo( true ) );
 
         File downloaded = new File( downloads, path );
         assertThat( "File: " + path + " doesn't seem to have been downloaded!", downloaded.exists(), equalTo( true ) );
-        assertThat( "Downloaded file: " + path + " contains the wrong content!", FileUtils.readFileToByteArray( downloaded ), equalTo( content ) );
+        assertThat( "Downloaded file: " + path + " contains the wrong content!",
+                    FileUtils.readFileToByteArray( downloaded ), equalTo( content ) );
+
+        //re-run to test the function of avoiding re-downloading existing files
+        Main aviodedMain = run( opts );
+        assertThat( "Wrong number of downloads logged. Should have been 0.", aviodedMain.getDownloaded(),
```

Comment:
This is going to seem redundant, but can we copy this test, give it a name that describes the avoided-download logic, and then add this second-run logic to that? It would be useful to see both tests run, and if we ever have a problem that only related to avoided logic, one would succeed...which immediately helps isolate the problem before debugging even starts.